app/src/androidTest/java/com/github/orkest/ViewModel/DeezerApiIntegrationTest.kt

```diff
@@ -337,5 +350,55 @@ class DeezerApiIntegrationTest {
         assertThrows(java.util.concurrent.CompletionException::class.java){mockDeezerIntegration.addANewSongToOrkestPlayList("dummy","", "").join()}
     }
 
+    /**
+     * Test for Deezer Welcome Activity
+     *
+     */
+
+    @get:Rule
+    var composeTestRule =  createComposeRule()
+
+
+    @Test
+    fun DeezerUi(){
+        composeTestRule.setContent {
+            DeezerWelcomeActivity(true).CreateViewForDeezer()
+        }
+        val button =composeTestRule.onNodeWithText("Start to use Deezer")
+        button.assertExists()
+        button.performClick()
+        composeTestRule.onNodeWithText("Feed").assertIsDisplayed()
+
+    }
+
+
+
+    @Test
+    fun testDeezerWelcomeActivity(){
+        composeTestRule.setContent {
+            DeezerWelcomeActivity(true).CreateViewForDeezer()
+        }
+        val button =composeTestRule.onNodeWithText("Start to use Deezer")
+        button.assertExists()
+        button.performClick()
+        /*
+        val db = Firebase.firestore
+        db.useEmulator("10.0.2.2", 8080)
+        db.firestoreSettings = firestoreSettings {
+            isPersistenceEnabled = false
+        }
+
+         */
+
+
+        DeezerWelcomeActivity.WelcomeOperation("CodeVale",DeezerApiIntegration(DeezerMockAPi.DeezerMockApiImplemented()),"BOB").join()
+
+        val info = FireStoreDatabaseAPI().getUserDeezerInformations("BOB").get()
+        assertEquals(info.access_token,"CodeVale")
+        assertEquals(info.playlistId,"3645740262")
+        assertEquals(info.userId,"2297625024")
+        //
+    }
+
```

Comment:
Maybe have more explicit test names. testDeezerWelcomeActivity isn't very explicit.

You could also do some exception handling with this line for example : DeezerWelcomeActivity.WelcomeOperation("CodeVale",DeezerApiIntegration(DeezerMockAPi.DeezerMockApiImplemented()),"BOB").join()